### PR TITLE
Test building with meson branch

### PR DESCRIPTION
**Pull request by lesteve**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by lesteve** [comment]
For some reason defining the commands to run in meta.yaml cause what seems like a quoting issue, and moving them to a `build.sh` file avoids the issue:

Here was the error for later reference:
```
+ /home/conda/feedstock_root/build_artifacts/scikit-learn_1709907407136/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/python -m build -w -n -x ' -Cbuilddir=builddir' ' -Csetup-args=--prefix=/home/conda/feedstock_root/build_artifacts/scikit-learn_1709907407136/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol' -Csetup-args=-Dlibdir=lib ' '
usage: python -m build [-h] [--version] [--sdist] [--wheel] [--outdir PATH]
                       [--skip-dependency-check] [--no-isolation]
                       [--config-setting KEY[=VALUE]]
                       [srcdir]
python -m build: error: unrecognized arguments:  -Csetup-args=--prefix=/home/conda/feedstock_root/build_artifacts/scikit-learn_1709907407136/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol  
+ cat builddir/meson-logs/meson-log.txt
```

File: recipe/build.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+# HACK: extend $CONDA_PREFIX/meson_cross_file that's created in
+# https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/main/recipe/activate-gcc.sh
+# https://github.com/conda-forge/clang-compiler-activation-feedstock/blob/main/recipe/activate-clang.sh
+# to use host python; requires that [binaries] section is last in meson_cross_file
+echo "python = '${PREFIX}/bin/python'" >> ${CONDA_PREFIX}/meson_cross_file.txt
+
+# meson-python already sets up a -Dbuildtype=release argument to meson, so
+# we need to strip --buildtype out of MESON_ARGS or fail due to redundancy
+MESON_ARGS_REDUCED="$(echo $MESON_ARGS | sed 's/--buildtype release //g')"
```
Comment (ogrisel) [inline recipe/build.sh:15]: Where does the original `MESON_ARGS` come from? conda-build? or some default conda-forge specific config environment? Shall we report this problem upstream?

File: recipe/build.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+mkdir builddir
+# -wnx flags mean: --wheel --no-isolation --skip-dependency-check
+$PYTHON -m build -w -n -x \
+        -Cbuilddir=builddir \
+        -Csetup-args=${MESON_ARGS_REDUCED// / -Csetup-args=} \
```
Comment (ogrisel) [inline recipe/build.sh:21]: I don't understand this line. I am not familiar with the forward slash syntax in that context. What do the do?

File: recipe/build.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+set -ex
+
+# necessary for cross-compilation to point to the right env
+export PKG_CONFIG_PATH=$PREFIX/lib/pkgconfig
+
+# HACK: extend $CONDA_PREFIX/meson_cross_file that's created in
+# https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/main/recipe/activate-gcc.sh
+# https://github.com/conda-forge/clang-compiler-activation-feedstock/blob/main/recipe/activate-clang.sh
+# to use host python; requires that [binaries] section is last in meson_cross_file
+echo "python = '${PREFIX}/bin/python'" >> ${CONDA_PREFIX}/meson_cross_file.txt
```
Comment (ogrisel) [inline recipe/build.sh:11]: Do other projects like scipy need similar hacks? Should this problem be reported/cross linked with issues in the `*-compiler-activation-feedstock` repos?

File: recipe/build.sh (hunk 2 above)
Comment (lesteve) [reply recipe/build.sh:21]: This add -Csetup-args= in front of every arguments in `MESON_ARGS_REDUCED`. This must be an advanced bash variable replacement

```
❯ echo $MESON_ARGS_REDUCED
--prefix=/home/conda/feedstock_root/build_artifacts/scikit-learn_1709907407136/prefix -Dlibdir=lib

```
❯ echo ${MESON_ARGS_REDUCED// / -Csetup-args=}
--prefix=/home/conda/feedstock_root/build_artifacts/scikit-learn_1709907407136/prefix -Csetup-args=-Dlibdir=lib
````

File: recipe/build.sh (hunk 3 above)
Comment (lesteve) [reply recipe/build.sh:11]: Yes I took it from numpy and scipy feedstock

File: recipe/build.sh (hunk 1 above)
Comment (lesteve) [reply recipe/build.sh:15]: `MESON_ARGS` comes from conda-forge: https://conda-forge.org/docs/maintainer/knowledge_base/#how-to-enable-cross-compilation

File: recipe/build.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+# meson-python already sets up a -Dbuildtype=release argument to meson, so
+# we need to strip --buildtype out of MESON_ARGS or fail due to redundancy
```
Comment (ogrisel) [inline recipe/build.sh:14]:
```suggestion
# conda-forge itself prepropulates MESON_ARGS for cross-compilation with meson:
# https://conda-forge.org/docs/maintainer/knowledge_base/#how-to-enable-cross-compilation
# meson-python already sets up a -Dbuildtype=release argument to meson, so
# we need to strip --buildtype out of MESON_ARGS or fail due to redundancy
```

File: recipe/build.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+MESON_ARGS_REDUCED="$(echo $MESON_ARGS | sed 's/--buildtype release //g')"
+
+mkdir builddir
+# -wnx flags mean: --wheel --no-isolation --skip-dependency-check
```
Comment (ogrisel) [inline recipe/build.sh:18]:
```suggestion
# -wnx flags mean: --wheel --no-isolation --skip-dependency-check
# Forward MESON_ARGS_REDUCED to our Python build tool by prepending -Csetup-args=
# in front of every arguments in MESON_ARGS_REDUCED.
```

**Review by ogrisel — APPROVED** [review]
LGTM.

**Comment by jeremiedbb** [comment]
done for 1.5.0rc1 in https://github.com/conda-forge/scikit-learn-feedstock/pull/258

**Pull request closed without merging**